File: backend/src/data/entity/UserEntity.ts
class UserEntity implements Entity {
    userId: string;
    firstName: string;
    lastName: string;
    userName: string;
    email: string;
    phoneNumber: string;
    password: string;
    createdAt: string;
    updatedAt: string;
    avatar: string;
    gender: number;
    isActive: boolean;
    dob: string;
    permisstionId: string;

    constructor(userId: string,
        firstName: string,
        lastName: string,
        userName: string,
        email: string,
        phoneNumber: string,
        password: string,
        createdAt: string,
        updatedAt: string,
        avartar: string,
        gender: number,
        isActive: boolean,
        dob: string,
        permissionId: string) {

        this.userId = userId;
        this.firstName = firstName;
        this.lastName = lastName;
        this.userName = userName;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.password = password;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.avatar = avartar;
        this.isActive = isActive;
        this.dob = dob;
        this.permisstionId = permissionId;
        this.gender = gender;
    }
}

File: android-app/data/src/main/java/com/starlight/module/data/datasource/entity/UserEntity.kt
package com.starlight.module.data.datasource.entity

import com.google.gson.annotations.SerializedName

class UserEntity(
    @SerializedName("user_name")
    val userName: String,
    @SerializedName("user_id")
    val userId: String,
    @SerializedName("first_name")
    val firstName: String,
    @SerializedName("last_name")
    val lastName: String,
    @SerializedName("email")
    val email: String,
    @SerializedName("phone_number")
    val phoneNumber: String,
    @SerializedName("is_active")
    val isActive: String,
    @SerializedName("avatar")
    val avatar: String?,
    @SerializedName("gender")
    val gender: Int,
    @SerializedName("dob")
    val dob: String?,
    @SerializedName("permission_type")
    val permissionType: Int
)